Content/Items/Gravedigger/Weapons.StaffOfSelfDissassembly.cs

```diff
@@ -0,0 +1,328 @@
+﻿using StarlightRiver.Content.Buffs;
+using System;
+using System.Linq;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Gravedigger
+{
+	internal class StaffOfSelfDissassembly : ModItem
+	{
+		public override string Texture => AssetDirectory.GravediggerItem + Name;
+
+		public override void Load()
+		{
+			StarlightPlayer.PreUpdateBuffsEvent += UpdateMinions;
+		}
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Staff of Self Dissassembly");
+			Tooltip.SetDefault("Reserve 40 life to summon a flesh apparation\n" +
+				"apparations grant increased life regeneration\n" +
+				"life regeneration boost is increased when apparitions strike an enemy");
+		}
+
+		public override void SetDefaults()
+		{
+			Item.damage = 60;
+			Item.DamageType = DamageClass.Summon;
+			Item.rare = ItemRarityID.Orange;
+			Item.useStyle = ItemUseStyleID.Swing;
+			Item.useAnimation = 20;
+			Item.useTime = 20;
+			Item.shoot = ModContent.ProjectileType<FleshApparation>();
+			Item.shootSpeed = 1;
+		}
+
+		public override bool CanUseItem(Player player)
+		{
+			return player.GetModPlayer<ResourceReservationPlayer>().TryReserveLife(40);
+		}
+
+		public override bool? UseItem(Player player)
+		{
+			player.AddBuff(ModContent.BuffType<FleshApparationBuff>(), 1800);
```

Comment:
Due to the minions behavior this has minimal effect on gameplay, and changing it could lead to strange behavior itself so I think its best to keep it this way